### PR TITLE
En 6169 system tests

### PR DESCRIPTION
**Pull request by iulianpascalau**
- fixes after system tests
- made all nodes register on the metachain directly (just as they would be directly staked)
- fixed some counters, added prints, fixed scripts
- changed the way genesis initial accounts are checked against initial nodes

### CODE REVIEW

File: genesis/process/shardGenesisBlockCreator.go [hunk 1]
```diff
@@ -494,3 +511,35 @@ func executeDelegation(
 
 	return delegationProcessor.ExecuteDelegation()
 }
+
+func incrementCrossShardDelegators(processors *genesisProcessors, arg ArgsGenesisBlockCreator) (int, error) {
```
Comment (AdoAdoAdo) [inline genesis/process/shardGenesisBlockCreator.go:515]: 1. The name of the function is misleading.
I would expect a crossShardDelegator++
what it does besides returing the number of delegations, is incrementing the nonces for each of the accounts that do delegation, for every delegation they do cross shad. 

so I would rename the function to something like updateNoncesForCrossShardDelegations

2. What happens with the nonces for accounts that do delegations in the same shard?

File: genesis/process/shardGenesisBlockCreator.go [hunk 2]
```diff
@@ -453,22 +460,32 @@ func deployInitialSmartContract(
 	return deployProc.Deploy(sc)
 }
 
-func incrementStakersNonces(arg ArgsGenesisBlockCreator) (int, error) {
+func incrementStakersNonces(processors *genesisProcessors, arg ArgsGenesisBlockCreator) (int, error) {
```
Comment (AdoAdoAdo) [inline genesis/process/shardGenesisBlockCreator.go:463]: Is it still one nonce increase if for example one account does staking for multiple nodes?

File: genesis/process/intermediate/delegationDeployProcessor.go [hunk 3]
```diff
@@ -39,11 +45,12 @@ func NewDelegationDeployProcessor(
...
 
 	ddp := &delegationDeployProcessor{
-		deployProcessor: deployProcessor,
+		DeployProcessor: deployProcessor,
 		accountsParser:  accountsParser,
+		pubkeyConv:      pubkeyConv,
 		nodePrice:       nodePrice,
 	}
-	ddp.replacePlaceholders = ddp.replaceDelegationPlaceholders
+	ddp.SetReplacePlaceholders(ddp.replaceDelegationPlaceholders)
```
[the remark this answers is not in the document]
Comment (iulianpascalau) [reply genesis/process/intermediate/delegationDeployProcessor.go:53]: we have an init parameter string for some SC, including the delegation SC. These strings can not be hardcoded in the json config as we might not know that value precisely. For example, in the delegation SC we have the total number of staking value and a sc of the delegation. These things need to be computed. If you take a look how a genesisSmartContract.json is being produced on the elrond-go-deploy project, you would see a string like: `%sc_total_stake%@1000@%auction_sc_address%` those values borded by the `%` character is a placeholder. that is why the delegationDeployProcessor has some extra function to to replace its custom placeholders.

File: cmd/node/main.go [hunk 4]
```diff
@@ -1292,16 +1292,21 @@ func startNode(ctx *cli.Context, log logger.Logger, version string) error {
...
-	log.Debug(
-		"restarting node",
-		"reason",
-		endProcessArgument.Reason,
-		"description",
-		endProcessArgument.Description,
-	)
 	switch endProcessArgument.Reason {
 	case core.ShuffledOut:
+		log.Debug(
```
[the remark this answers is not in the document]
Comment (iulianpascalau) [reply cmd/node/main.go:1297]: reason is not populated when a normal close occur: we will end up having something like "closing node    reason =   description = "
placed `log.Debug("closing node")` on top of this function

File: genesis/parsing/accountsParser.go [hunk 5]
```diff
@@ -1,6 +1,7 @@
 package parsing
 
 import (
+	"bytes"
 	"fmt"
 	"math/big"
```
[the remark this answers is not in the document]
Comment (iulianpascalau) [reply genesis/parsing/accountsParser.go:7]: done

File: genesis/process/intermediate/delegationDeployProcessor.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (iulianpascalau) [reply genesis/process/intermediate/delegationDeployProcessor.go:53]: Regarding < 0, done but there is no way that value can be below 0

File: genesis/process/metaGenesisBlockCreator.go [hunk 6]
```diff
@@ -365,6 +365,9 @@ func deploySystemSmartContracts(
 }
 
 // setStakedData sets the initial staked values to the staking smart contract
+// it will register both categories of nodes: direct staked and delegated stake. This done because it is the only
```
[the remark this answers is not in the document]
Comment (iulianpascalau) [reply genesis/process/metaGenesisBlockCreator.go:368]: fixed

File: cmd/node/main.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (AdoAdoAdo) [reply cmd/node/main.go:1297]: ah, ok, I thought that it should be populated

File: genesis/process/shardGenesisBlockCreator.go (hunk 1 above)
Comment (iulianpascalau) [reply genesis/process/shardGenesisBlockCreator.go:515]: renamed to incrementNoncesForCrossShardDelegations (as it does an increment). The reason this is not called for the intra shard delegators is that the tx processor does that automatically when executing the transaction

File: genesis/process/shardGenesisBlockCreator.go (hunk 2 above)
Comment (iulianpascalau) [reply genesis/process/shardGenesisBlockCreator.go:463]: good point. will change.

Comment (iulianpascalau) [reply genesis/process/shardGenesisBlockCreator.go]: done

**Pull request merged**